test/adapters/redis_test.rb

```diff
@@ -0,0 +1,15 @@
+require 'test_helper'
+require 'flipper/adapters/redis'
+
+class RedisTest < MiniTest::Test
+  prepend SharedAdapterTests
+
+  def setup
+   options = {}
+   if ENV['BOXEN_REDIS_URL']
```

Comment:
*I actually don't use boxen anymore*
great, will remove it in this PR

*I'd like this to live under lib/ as the shared adapter specs do*
Awesome makes sense, will change that, thanks

*Why use prepend when we could use include? Include should work the same here and I've not really used prepend so I'm just curious.*

It comes down to where we want to call super.  I think it's nice to just expect someone using the shared_adapter_tests to only have to define a setup method that initializes an @adapter and effectively hide some of the implementation details.  I'll give an example of difference between `prepend` and `include`.

`include`
class RedisTest < MiniTest::Test
  include SharedAdapterTests

  def setup
   client = Redis.new({}).tap { |c| c.flushdb }
   @adapter = Flipper::Adapters::Redis.new(client)
    super
  end
end

using `include` will make RedisTest#setup take precedence over the module's `setup` so in this case minitest calls RedisTest.new.setup, which initializes @adapter and then calls super to look up the inheritance chain and call the module's setup.


`prepend` makes the prepended module's `setup` take precedence over RedisTest#setup, so when minitest calls RedisTest.new.setup, the module's `setup` will be called first, calling super which calls RedisTest#setup and initializes @adapter then hands back control to the module to run all the tests.